src/main/java/uk/gov/hmcts/ccd/domain/service/common/CaseDataService.java

```diff
@@ -49,13 +51,19 @@ private void deduceDefaultClassifications(final JsonNode dataNode,
                     final String caseFieldType = caseField.getFieldType().getType();
                     if (caseFieldType.equalsIgnoreCase(COMPLEX_TYPE)) {
                         found = true;
-                        deduceClassificationForComplexType(dataNode, fieldIdPrefix, fieldName, caseField);
+                        deduceClassificationForComplexType(dataNode,
+                                                           getExistingDataClassificationNodeOrEmpty(existingDataClassificationNode, fieldName), fieldIdPrefix, fieldName, caseField);
+                        break;
```

Comment:
Performance improvement in that we do not need to loop over all case fields once we found the searched for one.